core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1289,13 +1289,16 @@ public Optional<TableHandle> applyDelete(Session session, TableHandle table)
     }
 
     @Override
-    public OptionalLong executeDelete(Session session, TableHandle table)
+    public OptionalLong executeDelete(Session session, TableHandle table, List<TableHandle> sourceTableHandles)
     {
         CatalogHandle catalogHandle = table.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
         ConnectorSession connectorSession = session.toConnectorSession(catalogHandle);
-
-        return metadata.executeDelete(connectorSession, table.getConnectorHandle());
+        List<ConnectorTableHandle> sourceConnectorHandles = sourceTableHandles.stream()
+                .filter(handle -> handle.getCatalogHandle().equals(catalogHandle))
```

Comment:
Any reason why we wan't only handles from the same catalog?

Reply:
This is intentional in order to ensure that schema and table name are pointing to the very same table.

If we'd have `catalog1.schema1.table1` and `catalog2.schema1.table1` we wouldn't be able to tell whether these tables are one and the same (because they may point to completely different tables even though their schema and table name are the same).